app/components/autocomplete/at_mention/index.ts

```diff
@@ -26,21 +28,48 @@ const enhanced = withObservables([], ({database, channelId}: WithDatabaseArgs &
 
     let useChannelMentions: Observable<boolean>;
     let useGroupMentions: Observable<boolean>;
+    let isChannelConstrained: Observable<boolean>;
+    let isTeamConstrained: Observable<boolean>;
+    let team: Observable<TeamModel | undefined>;
 
     if (channelId) {
         const currentChannel = observeChannel(database, channelId);
+        team = currentChannel.pipe(switchMap((c) => {
+            if (c && c.teamId) {
+                const mapTeam = (t: TeamModel | null) => (t ? t.observe() : of$(undefined));
+                return c.team.observe().pipe(switchMap(mapTeam));
```

Comment:
@enahum / @larkox - I've changed it so I don't have to type cast it anymore; does that work better?

Reply:
you could actually use `c?.teamId ? observerTeam(database, c.teamId) : observeCurrentTeam(database)`

Reply:
This causes issues further down when I'm passing though the model, where it expects it to be `Model|undefined`, and those are inferred from the `Observable<Model...` so I'd have to do a lot of changing :(